### DISCUSSION TITLE
Enhanced timemap info: Include element attr info

### DISCUSSION THREAD

**Discussion by andre2007** [rism-digital/verovio#3848] [Q&A] [answered]
Hi,

In my use case, I would need to call for every note ID the toolkit method `getElementAttr` to get the `oct` and `pname`.
Maybe an option could be added to the toolkit method `renderToTimemap` to already include these information in the output.

```
string s = toolkit.renderToTimemap(`{"includeElementAttr": true}`)
```

Kind regards
André

**Comment by lpugin** [comment, marked as answer]
These functions are all meant to be as generic as possible. As a result, more than one call might be needed. Adding more data could be possible, but the problem is that your problem need this, but the next project will need that, and so on.

However, you do not necessarily need to use the toolkit method to retrieve element attributes one by one. You can get them from the MEI data directly. If you are loading another format into Verovio, you can use `getMEI` to obtain it in one go. That would be quite similar to have then in the timemap.

**Comment by andre2007** [comment]
Thanks. 

I just checked and it is working to get the same data using mei. 
You are correct, my use case is too specific to add these data to the timemap.

However, I wonder whether it make sense to provide a mass enabled function of getElementAttr?

Therefore, something like this:

``` d
string s = toolkit.getElementsAttr("xmlId1,xmlId2,xmlId3");
```

which returns a json like this:

``` json
{
    "xmlId1": {
        "dur": "4",
        "dur.ppq": "1",
        "oct": "4",
        "pname": "f"
    }, 
    "xmlId2": {
        "dur": "4",
        "dur.ppq": "1",
        "oct": "4",
        "pname": "f"
    }, 
}
```
